src/linker/configuration.py

```diff
@@ -29,6 +33,8 @@ def __init__(
         self.computing_environment = self.environment["computing_environment"]
         self.container_engine = self.environment.get("container_engine", "undefined")
         self.spark = self.environment.get("spark", None)
+        self.schema = self._get_schema()
+        self.implementations = self._get_implementations()
```

Comment:
We can't call this before validating the structure of the pipeline configuration, as it may be that there is no value at `self.pipeline["steps"][step_name]["implementation"]`

Reply:
I think that vulnerability predates this PR, but I don't disagree